SDK upgrade causes compile failure due to not generating VB.NET WPF code behind

Repro:

Clone git@github.com:dotnet/project-system.git.

- Build with SDK 3.1.400-preview-015178 ✔️ 
- Build with SDK 5.0.100-preview.6.20318.15 ❌ 

Error:

> src\Microsoft.VisualStudio.Editors\OptionPages\GeneralOptionPageControl.xaml.vb(45,13): error BC30451: 'InitializeComponent' is not declared. It may be inaccessible due to its protection level.

The project file defines the item as `Page` but the Properties pane shows it as `None`.

Potentially bad globs?

Note that this is a VB.NET file.

Also note that this worked with an earlier version of 5.0.100-preview*, but I don't know which as it was uninstalled when I installed preview 6. I _think_ it was preview 4.

Doing some quick testing (thanks @Forgind) it doesn't look like that suggestion works @dsplaisted . I'm thinking at this point we make this opt-in instead of opt-out. So if `ImportWinFXTargets` isn't set, SDK sets it to true by default. This reverts to functionality before the [original issue](https://github.com/dotnet/msbuild/issues/4948) and provides a workaround for those that run into that. How does this sound?

An interesting observation is that when we force `ImportFrameworkWinFXTargets` to true, we get a different issue in [this repro](https://github.com/dotnet/sdk/issues/12739). @davkean is this something you've seen before? This error happens on 16.8.0 Preview 2.0 [30404.33.master]. It looks unrelated to project system since it works in 16.6.5.

Error:
```
C:\Users\namytelk\Documents\GitHub\project-system\src\Microsoft.VisualStudio.ProjectSystem.Managed.VS\ProjectSystem\VS
\Waiting\VisualStudioWaitIndicator.cs(101,24): error CS8619: Nullability of reference types in value of type '(WaitInd
icatorResult Canceled, T?)' doesn't match target type '(WaitIndicatorResult, T)'. [C:\Users\namytelk\Documents\GitHub\
project-system\src\Microsoft.VisualStudio.ProjectSystem.Managed.VS\Microsoft.VisualStudio.ProjectSystem.Managed.VS_q44
qb5gw_wpftmp.csproj]
C:\Users\namytelk\Documents\GitHub\project-system\src\Microsoft.VisualStudio.ProjectSystem.Managed.VS\ProjectSystem\VS
\Waiting\VisualStudioWaitIndicator.cs(108,24): error CS8619: Nullability of reference types in value of type '(WaitInd
icatorResult Canceled, T?)' doesn't match target type '(WaitIndicatorResult, T)'. [C:\Users\namytelk\Documents\GitHub\
project-system\src\Microsoft.VisualStudio.ProjectSystem.Managed.VS\Microsoft.VisualStudio.ProjectSystem.Managed.VS_q44
qb5gw_wpftmp.csproj]
```

--- a/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.BeforeCommon.targets
+++ b/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.BeforeCommon.targets
@@ -25,7 +25,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     Make it opt-out, just in case, if something fails or we don't want to use 'WindowsDesktop' SDK.
   -->
   <PropertyGroup>
-    <ImportFrameworkWinFXTargets Condition="'$(ImportFrameworkWinFXTargets)' == ''">false</ImportFrameworkWinFXTargets>
+    <ImportFrameworkWinFXTargets Condition="'$(ImportFrameworkWinFXTargets)' == '' and '$(TargetFrameworkIdentifier)' != '.NETFramework'">false</ImportFrameworkWinFXTargets>
   </PropertyGroup>
 
   <PropertyGroup>

--- a/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.BeforeCommon.targets
+++ b/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.BeforeCommon.targets
@@ -20,12 +20,10 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <!--
     Workaround: https://github.com/microsoft/msbuild/issues/4948
-    Disable .NET Framework's inbox WinFX targets when using the SDK, since, we really don't use it's build logic
-    and is superseded by 'WindowsDesktop' SDK that provides it's own WinFX for both NETFX and CoreCLR targets.
-    Make it opt-out, just in case, if something fails or we don't want to use 'WindowsDesktop' SDK.
+    Allow opt-out of importing framework's winfx.targets.
   -->
   <PropertyGroup>
-    <ImportFrameworkWinFXTargets Condition="'$(ImportFrameworkWinFXTargets)' == '' and '$(TargetFrameworkIdentifier)' != '.NETFramework'">false</ImportFrameworkWinFXTargets>
+    <ImportFrameworkWinFXTargets Condition="'$(ImportFrameworkWinFXTargets)' == ''">true</ImportFrameworkWinFXTargets>
   </PropertyGroup>
 
   <PropertyGroup>

